app/connection/StreamManager.js

```diff
@@ -41,7 +41,9 @@ class StreamManager {
         // It is not its responsibility to reestablish the streams upon reconnection.
         this._streamIds = {}; // set it to blank so that forget requests do not get called
         for (const key of Object.keys(this._streams)) {
-            this._forgetStream(key);
+            if (this._streams[key].subscriberCount !== 0) {
```

Comment:
we shouldn't need this. Forget requests do not get sent when `this._streamIds` is set to `{}`. The code block here simply cleans up the unused streams from memory.

Reply:
I was able to get it working without `if (this._streams[key].subscriberCount !== 0) { }`

Reply:
@MahboobehMohammadi the forgetrequest is beign send here. cause we removed the stream.
https://github.com/binary-com/SmartCharts/blob/1003dfe3861a7f785691b137e024c348e0ba44a7/app/connection/StreamManager.js#L25